web/app/components/workflow/nodes/_base/components/variable/var-reference-picker.tsx

```diff
@@ -102,19 +103,8 @@ const VarReferencePicker: FC<Props> = ({
 
   const [varKindType, setVarKindType] = useState<VarKindType>(defaultVarKindType)
   const isConstant = isSupportConstantValue && varKindType === VarKindType.constant
-  const outputVars = useMemo(() => {
-    if (availableVars)
-      return availableVars
-
-    const vars = getNodeAvailableVars({
```

Comment:
It seems that there is a bug in getNodeAvailableVars. Why not fix the problem of getNodeAvailableVars and use useAvailableVarList directly? Because there are other places in the project that use getNodeAvailableVars.

Reply:
There is no error in `getNodeAvailableVars` itself.
The issue arises from it being wrapped in` useMemo`, causing the variable list not to refresh correctly. A simple solution is to remove useMemo.
 However, since I noticed the same logic in `useAvailableVarList`, I opted to use useAvailableVarList instead.

Reply:
> It seems that there is a bug in getNodeAvailableVars. Why not fix the problem of getNodeAvailableVars and use useAvailableVarList directly? Because there are other places in the project that use getNodeAvailableVars.

There is no error in getNodeAvailableVars itself.
The issue arises from it being wrapped in useMemo, causing the variable list not to refresh correctly. A simple solution is to remove useMemo.
However, since I noticed the same logic in useAvailableVarList, I opted to use useAvailableVarList instead.

Reply:
Great job! I will check it later.